flexx/ui/pywidgets/_dynamicwidgetcontainer.py

```diff
@@ -0,0 +1,81 @@
+from ... import event
+from .._widget import PyWidget
+import threading
+
+
+class DynamicWidgetContainer(PyWidget):
+    """ Widget container to allow dynamic insertion and disposal of widgets.
+    """
+
+    DEFAULT_MIN_SIZE = 0, 0
+
+    def init(self, *init_args, **property_values):
+        # TODO: figure out if init_args is needed for something
+        super(DynamicWidgetContainer, self).init()  # call to _component.py -> Component.init(self)
+        # the page
+        self.pages = []  # pages are one on top of another
+
+    def _init_events(self):
+        pass  # just don't use standard events
+
+    def clean_pages(self):  # remove empty pages from the top
+        while self.pages[-1] == None:
+            del self.pages[-1]
+
+    @event.reaction("remove")
+    def __remove(self, *events):
+        if self.pages:
+            page = self.pages[events[0]['page_position']]
+            page.dyn_stop_event.set()
+            page.dyn_id = None
+            page.dispose()
+            page._jswidget.dispose()  # <-- added
+            self.pages[events[0]['page_position']] = None
+
+    @event.emitter
+    def remove(self, page_position):
+        return dict(page_position=page_position)
+
+    @event.reaction("_emit_instantiate")
+    def __instantiate(self, *events):
+        with self:
+            with events[0]['widget_type'](events[0]['style']) as page:
```

Comment:
Could you please add a few lines of comments explaining the flow. As I understand it, first a `threading.task` is appended to `.pages` and that is exchanged for an actual page instance here. Does this trick have to do with thread-safety?